### PR TITLE
[Merged by Bors] - feat(NormNum): add extension for `Nat.ModEq`, `Int.ModEq`

### PR DESCRIPTION
**Pull request by yury-harmonic**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — APPROVED** [review]

**Comment by grunweg** [comment]
Looks fine at first glance, but I don't know the fine points too well.
@eric-wieser Would you like to take a look? And sorry for asking you for so many reviews recently...

File: Mathlib/Tactic/NormNum/ModEq.lean [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace Mathlib.Meta.NormNum
+
+open Qq
+
+/-- `norm_num` extension for `Nat.ModEq`. -/
+@[norm_num Nat.ModEq _ _ _]
+def evalNatModEq : NormNumExt where eval {u αP} e := do
+  match u, αP, e with
+  | 0, ~q(Prop), ~q(Nat.ModEq $n $a $b) =>
```
Comment (eric-wieser) [inline Mathlib/Tactic/NormNum/ModEq.lean]:
```suggestion
  | 0, ~q(Prop), ~q($a ≡ $b [MOD $n]) =>
```
etc

File: Mathlib/Tactic/NormNum/ModEq.lean [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+In this file we define `norm_num` extensions for `a ≡ b [MOD n]` and `a ≡ b [ZMOD n]`.
+-/
+
+namespace Mathlib.Meta.NormNum
+
+open Qq
+
+/-- `norm_num` extension for `Nat.ModEq`. -/
+@[norm_num Nat.ModEq _ _ _]
```
Comment (eric-wieser) [inline Mathlib/Tactic/NormNum/ModEq.lean]:
```suggestion
@[norm_num _ ≡ _ [MOD _]]
```
etc

File: MathlibTest/norm_num_ext.lean [hunk 3]
```diff
@@ -478,6 +479,13 @@ example : ¬ (2 : ℤ) ∣ 5 := by norm_num1
 example : (553105253 : ℤ) ∣ 553105253 * 776531401 := by norm_num1
 example : ¬ (553105253 : ℤ) ∣ 553105253 * 776531401 + 1 := by norm_num1
 
+example : 10 ≡ 7 [MOD 3] := by norm_num1
+example : ¬ (10 ≡ 7 [MOD 5]) := by norm_num1
+example : 10 ≡ 7 [ZMOD 3] := by norm_num1
+example : ¬ (10 ≡ 7 [ZMOD 5]) := by norm_num1
+example : -3 ≡ 7 [ZMOD 5] := by norm_num1
+example : ¬ (-3 ≡ 7 [ZMOD 50]) := by norm_num1
```
Comment (eric-wieser) [inline MathlibTest/norm_num_ext.lean:488]: Could you add a test with some negative values?

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Comment by yury-harmonic** [comment]
bors merge
